### PR TITLE
chore: minor evaluator clean up

### PR DESCRIPTION
**Pull request by sash-a**
## What?
Minor clean up to the evaluator. 

Removed the creation of trained params and eval keys from the eval setup method as it's not really the responsibility of that method, it should just make the evaluation methods.

Also created the `unreplicate_batch_dim` method to easily unreplicate our `update_batch_size` dim.

Finally removed the unecessary vmap inside the evaluator as flax will automatically vmap for us

### CODE REVIEW

File: mava/systems/ff_ippo.py [hunk 1]
```diff
@@ -464,16 +471,12 @@ def run_experiment(_config: DictConfig) -> None:
...
     # Setup evaluator.
-    evaluator, absolute_metric_evaluator, (trained_params, eval_keys) = evaluator_setup(
-        eval_env=eval_env,
-        key_e=key_e,
-        network=actor_network,
-        params=learner_state.params.actor_params,
-        config=config,
-    )
+    # Broadcast trained params to cores and split keys for each core.
+    trained_params = unreplicate_batch_dim(learner_state.params.actor_params)
```
Comment (OmaymaMahjoub) [inline mava/systems/ff_ippo.py]: I don't think it is necessary to have trained params initialized before the for loop as we will get the first needed trained_params in line 533

File: mava/utils/jax.py [hunk 2]
```diff
@@ -59,3 +59,9 @@ def unreplicate_learner_state(
     and one for the `update batch size`.
     """
     return jax.tree_map(lambda x: x[(0,) * unreplicate_depth], learner_state)  # type: ignore
+
+
+def unreplicate_batch_dim(x: Any) -> Any:
+    """Unreplicated just the batch dim.
```
Comment (OmaymaMahjoub) [inline mava/utils/jax.py]: maybe specify that it is the update_batch_dim and getting the element 0 is fine as the params are replicated (broadcasted) over that batch

File: mava/systems/ff_ippo.py (hunk 1 above)
Comment (OmaymaMahjoub) [reply mava/systems/ff_ippo.py]: The same for all the other systems

**Review by OmaymaMahjoub — CHANGES_REQUESTED** [review]
Thanks @sash-a  for cleaning the evaluator :fire:  Just some minor comments!

File: mava/advanced_usage/ff_ippo_store_experience.py [hunk 3]
```diff
@@ -468,6 +472,9 @@ def run_experiment(_config: DictConfig) -> None:  # noqa: CCR001
     config = copy.deepcopy(_config)
     logger = MavaLogger(config)
 
+    # Calculate total timesteps.
```
Comment (RuanJohn) [inline mava/advanced_usage/ff_ippo_store_experience.py]: We are only getting the number of devices here, is this intended? Same for all systems.

File: mava/evaluator.py [hunk 4]
```diff
@@ -311,25 +307,12 @@ def evaluator_setup(
             10,
         )
     else:
-        vmapped_eval_apply_fn = jax.vmap(
-            network.apply,
-            in_axes=(None, 0),
-        )
-        evaluator = get_ff_evaluator_fn(eval_env, vmapped_eval_apply_fn, config, log_win_rate)
+        evaluator = get_ff_evaluator_fn(eval_env, network.apply, config, log_win_rate)
```
Comment (RuanJohn) [inline mava/evaluator.py:310]: I really like moving to network.apply and not having to vmap here. I just want us to be super sure that the networks are behaving as expected, ie. broadcasting over the agents. 
Did you run any performance tests on this @sash-a?

**Review by RuanJohn — COMMENTED** [review]
Overall looks very nice @sash-a! :clap: 
Just some small questions.

File: mava/evaluator.py (hunk 4 above)
Comment (OmaymaMahjoub) [reply mava/evaluator.py:310]: Nice point @RuanJohn , It may behave the same (@sash-a if you can test it and make sure this is true) and if this is the case something really important to check is steps per seconds

File: mava/systems/ff_ippo.py [hunk 5]
```diff
@@ -450,12 +454,15 @@ def run_experiment(_config: DictConfig) -> None:
     """Runs experiment."""
     config = copy.deepcopy(_config)
 
+    # Calculate total timesteps.
```
Comment (OmaymaMahjoub) [inline mava/systems/ff_ippo.py]: the comment is wrong here this is getting the number of devices and this comment should be before line 477 (same for other files)

File: mava/advanced_usage/ff_ippo_store_experience.py (hunk 3 above)
Comment (OmaymaMahjoub) [reply mava/advanced_usage/ff_ippo_store_experience.py]: yes because we need it before we initialize the evaluator as it is necessary for `eval_keys = jax.random.split(key_e, n_devices)`

Comment (sash-a) [reply mava/advanced_usage/ff_ippo_store_experience.py]: Ah my bad, copied the comment as well as the `n_devices` when I moved it

File: mava/evaluator.py (hunk 4 above)
Comment (sash-a) [reply mava/evaluator.py:310]: I remember testing this ages ago and it was the same, also I do this in SAC and it works fine. Although I'm actually a bit worried about it behaving incorrectly with CNNs 🤔

Comment (RuanJohn) [reply mava/evaluator.py:310]: I think a few simple ppo tests would be great for this if possible just to be sure please :pray:

Comment (sash-a) [reply mava/evaluator.py:310]: Testing...

**Review by OmaymaMahjoub — APPROVED** [review]
LGTM!

**Review by RuanJohn — APPROVED** [review]
:ship:
